src/ts/circular_buffer.ts

```diff
@@ -63,7 +63,15 @@ class circular_buffer<T> {
   }
 
 
-
+  fill(x:any): T[] {
```

Comment:
`x` doesn't take `any`, it takes `T`

`T` is the type that the container is constrained to.  So, fill should only take from that type

This is fine - you don't speak typescript yet